### ISSUE TITLE
fromBuffer and stream methods give different results for same ai file

### ISSUE THREAD

**Issue by chandern** [sindresorhus/file-type#426]
I recently found a couple of my tests broke because of changes in detecting ai files. I have a sample ai file that will give me file type "ai" using fromBuffer method but type "pdf" using stream method.  Before I would get both methods returning type "ai".
It would be nice if both methods return a consistent result.

test file
[test-ai.pdf](https://github.com/sindresorhus/file-type/files/5867580/test-ai.pdf)

**Comment by hdavidzhu** [comment]
(Related to https://github.com/sindresorhus/file-type/pull/396)

**Comment by Borewit** [comment]
```js
const fs = require('fs');

const FileType = require('file-type');
const path = require('path');

const fixture = path.join('fixture', 'test-ai.pdf');

(async () => {

  const fileType = await FileType.fromFile(fixture);
  console.log(`  file detected as: ext=${fileType.ext}, mime=${fileType.mime}`);

  const stream = fs.createReadStream(fixture);
  const streamType = await FileType.fromStream(stream);
  console.log(`stream detected as: ext=${streamType.ext}, mime=${streamType.mime}`);
  stream.close();

  const buffer = fs.readFileSync(fixture);
  const bufferType = await FileType.fromBuffer(buffer);
  console.log(`buffer detected as: ext=${bufferType.ext}, mime=${bufferType.mime}`);
})();
```

Outputs:
```
  file detected as: ext=ai, mime=application/postscript
stream detected as: ext=ai, mime=application/postscript
buffer detected as: ext=ai, mime=application/postscript
```

I am not able to reproduce the issue.

**Comment by Borewit** [comment]
_Out of issue scope comment:_

Noticed my master branch was out of sync, It took me a while figure out the default branch has been renamed.
Still need convince my IDE to deviate from the convention. 

Very confusing and annoying.

**Comment by chandern** [comment]
Thanks for looking in to this. I use the `.stream` method of file-type so I can use the stream further on in my application. Extending your example with how I use file-type I get the issue:

```
const fs = require('fs');

const FileType = require('file-type');
const path = require('path');

const fixture = path.join('fixture', 'test-ai.pdf');

(async () => {

  const fileType = await FileType.fromFile(fixture);
  console.log(`  file detected as: ext=${fileType.ext}, mime=${fileType.mime}`);

  const stream = fs.createReadStream(fixture);
  const streamType = await FileType.fromStream(stream);
  console.log(`stream detected as: ext=${streamType.ext}, mime=${streamType.mime}`);
  stream.close();

  const buffer = fs.readFileSync(fixture);
  const bufferType = await FileType.fromBuffer(buffer);
  console.log(`buffer detected as: ext=${bufferType.ext}, mime=${bufferType.mime}`);

  const altStream = fs.createReadStream(fixture);
  const fileTypeStream = await FileType.stream(altStream);
  console.log(`alt stream detected as: ext=${fileTypeStream.fileType.ext}, mime=${fileTypeStream.fileType.mime}`);
  altStream.close();
})();

```

Outputs:

```
file detected as: ext=ai, mime=application/postscript
stream detected as: ext=ai, mime=application/postscript
buffer detected as: ext=ai, mime=application/postscript
alt stream detected as: ext=pdf, mime=application/pdf
```

**Comment by Borewit** [comment]
Since version [v13.0.0](https://github.com/sindresorhus/file-type/releases/tag/v13.0.0) we moved away from the ~4 kilobyte sample boundary, allowing, if required, deeper analysis of files, streams or Buffers.

However `stream()`  is still limited to a 4k limit. That is what explains the difference. It understands it is kind of PostScript, but it required more data to understand it is `ai`.

I tried to get around this limitations in PR #300, but unfortunately this is not possible. 

Also related: https://github.com/sindresorhus/file-type/issues/341#issuecomment-604917968
